ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,51 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// invalid state: `{description}`
+    InvalidState { description: String },
+    /// missing state: `{description}`
+    MissingState { description: String },
+    /// failed to update store: `{description}`
+    FailedToStore { description: String },
+    /// failed to retrieve from store: `{description}`
+    FailedToRetrieve { description: String },
+    /// failed to parse data: `{description}`
+    FailedToParse { description: String },
```

Comment:
Took another look at where `FailedtoParse` is used, it seems those are handler errors, not from the host. Let’s just revert them to how they were and drop this variant too.